gravitee-node-api/src/main/java/io/gravitee/node/api/secrets/runtime/spec/Spec.java

```diff
@@ -0,0 +1,205 @@
+        }
+        ok = uri.charAt(0) == SecretURL.URL_SEPARATOR;
+        if (!ok) {
+            throw new IllegalArgumentException("uri must start with '%s'".formatted(SecretURL.URL_SEPARATOR));
+        }
+        if (name != null) {
+            assertName(name);
+        }
+    }
+
+    /**
+     * Return uri and key concatenated as specified in {@link Spec#formatUriAndKey(String, String)}
+     * @return a string concat of uri and key
+     */
+    public String uriAndKey() {
+        return formatUriAndKey(uri, key);
+    }
+
+    /**
+     * The spec converted to a SecretURL, it is then used to resolve the secret
+     * @return the spec as a SecretURL
+     */
+    public SecretURL toSecretURL() {
+        return SecretURL.from(uriAndKey(), false);
+    }
+
+    /**
+     * Simple string representation of unique data of the secret
+     * @return name if set or the uri
+     */
+    public String asSimpleString() {
+        return name != null && !name.isEmpty() ? name : uri;
+    }
+
+    /**
+     * Shortcut to {@link ACLs#fieldKind()}
+     * fieldKind or null
+     */
+    public FieldKind allowedFieldKind() {
+        return acls != null ? acls.fieldKind() : null;
+    }
+    /**
+     * Shortcut to extract all allowed fields from all {@link ACLs#plugins()} }
+     * @return a non-null set of allowed fields from ACLs object.
+     */
+    public Set<String> allowedFields() {
+        if (acls != null && acls.plugins() != null) {
+            return acls()
+                .plugins()
+                .stream()
+                .flatMap(pl -> {
+                    if (pl.fields() == null) {
+                        return Stream.empty();
+                    }
+                    return pl.fields().stream();
+                })
+                .map(String::toLowerCase)
+                .collect(Collectors.toSet());
+        }
+        return Set.of();
+    }
+
+    /**
+     * Checks if the Spec has the given resolution type.
+     * If {@link Resolution.Type#ONCE} is passed then it matched with a <code>null</code> resolution
+     * @param type type of resolution to check
+     * @return true of the type passed matches the one in the spec
+     */
+    public boolean hasResolutionType(@Nonnull Resolution.Type type) {
+        if (type == Resolution.Type.ONCE) {
+            return resolution == null || resolution.type() == Resolution.Type.ONCE;
+        }
+        return resolution != null && type.equals(resolution.type());
+    }
+
+    /**
+     * @return main uri and optionally key concat with {@link SecretURL#URI_KEY_SEPARATOR}
+     */
+    public static String formatUriAndKey(String uri, String key) {
+        if (key == null) {
+            return uri;
+        }
+        return uri.concat(SecretURL.URI_KEY_SEPARATOR).concat(key);
+    }
+
+    /*
+     * char by analysis to avoid using regexp on a user input
+     */
+    private void assertName(@Nonnull String name) {
```

Comment:
I feel like is too much complicate and could be simplify using a simple reg ex. Why do you want to avoid using one?

Reply:
I'll use io.gravitee.common.utils.IdGenerator to compare it is valid